cmake/Utils.cmake

```diff
@@ -252,3 +252,25 @@ function(add_check_gtest_target TARGET_NAME BINARY BINARY_DIR)
   )
   add_dependencies("check-gtest" ${TARGET_NAME})
 endfunction(add_check_gtest_target)
+
+macro(parse_cuda_version CUDA_VERSION CUDA_VERSION_MAJOR_VAR CUDA_VERSION_MINOR_VAR CUDA_VERSION_PATCH_VAR CUDA_VERSION_SHORT_VAR CUDA_VERSION_SHORT_DIGIT_ONLY_VAR)
```

Comment:
Can you make this a `function`? I'd say we should be like in C++: prefer functions over macros

Reply:
will do